gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobufeditablelayer.cpp

```diff
@@ -0,0 +1,230 @@
+    OGRGeometry* poFilterGeomBak = poEditableLayer->GetSpatialFilter();
+    if( poFilterGeomBak )
+        poFilterGeomBak = poFilterGeomBak->clone();
+    poEditableLayer->SetSpatialFilter(nullptr);
+
+    auto aoMapSrcToTargetIdx = poFlatGeobufTmpLayer->GetLayerDefn()->
+        ComputeMapForSetFrom(poEditableLayer->GetLayerDefn(), true);
+    aoMapSrcToTargetIdx.push_back(-1); // add dummy entry to be sure that .data() is valid
+
+    for( auto&& poFeature: poEditableLayer )
+    {
+        if( eErr != OGRERR_NONE )
+            break;
+        OGRFeature *poNewFeature =
+            new OGRFeature(poFlatGeobufTmpLayer->GetLayerDefn());
+        poNewFeature->SetFrom(poFeature.get(), aoMapSrcToTargetIdx.data(), true);
+        eErr = poFlatGeobufTmpLayer->CreateFeature(poNewFeature);
+        delete poNewFeature;
+    }
+    delete poFlatGeobufTmpLayer;
+
+    // Restore filters.
+    poEditableLayer->SetAttributeFilter(pszQueryStringBak);
+    CPLFree(pszQueryStringBak);
+    poEditableLayer->SetSpatialFilter(iFilterGeomIndexBak, poFilterGeomBak);
+    delete poFilterGeomBak;
+
+    if( eErr != OGRERR_NONE )
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "Error while creating %s",
+                 osTmpFilename.c_str());
+        VSIUnlink(osTmpFilename);
+        return eErr;
+    }
+
+    delete m_poFlatGeobufLayer;
+    *ppoDecoratedLayer = nullptr;
+    m_poFlatGeobufLayer = nullptr;
+
+    if( osFilename != osTmpFilename )
+    {
+        const CPLString osTmpOriFilename(osFilename + ".ogr_bak");
+        if (VSIRename(osFilename, osTmpOriFilename) != 0 ||
+            VSIRename(osTmpFilename, osFilename) != 0)
+        {
+            CPLError(CE_Failure, CPLE_AppDefined, "Cannot rename files");
+            return OGRERR_FAILURE;
+        }
+        VSIUnlink(osTmpOriFilename);
+    }
+
+    VSILFILE *fp = VSIFOpenL(osFilename, "rb+");
+    if( fp == nullptr)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "Cannot reopen updated %s",
+                 osFilename.c_str());
+        return OGRERR_FAILURE;
+    }
+
+    m_poFlatGeobufLayer = OGRFlatGeobufLayer::Open(osFilename.c_str(), fp, false, false);
+    *ppoDecoratedLayer = m_poFlatGeobufLayer;
+
+    return OGRERR_NONE;
+}
+
+OGRFlatGeobufEditableLayer::OGRFlatGeobufEditableLayer(OGRFlatGeobufLayer *poFlatGeobufLayer,
+                                         char **papszOpenOptions) :
+    OGREditableLayer(poFlatGeobufLayer, true,
+                     new OGRFlatGeobufEditableLayerSynchronizer(
+                         poFlatGeobufLayer, papszOpenOptions),
+                     true)
+{
+}
+
+OGRErr OGRFlatGeobufEditableLayer::CreateField( OGRFieldDefn *poNewField, int bApproxOK )
+{
+    if( m_poEditableFeatureDefn->GetFieldCount() >= 10000 )
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "Limiting to 10000 fields");
+        return OGRERR_FAILURE;
+    }
+
+    if( m_oSetFields.empty() )
```

Comment:
I believe everything m_oSetFields related was CSV specific. You could probably just remove it and the override of DeleteField() and AlterFieldDefn()